test/benchmarks/CMakeLists.txt

```diff
@@ -2,6 +2,7 @@ set(all_sources
   benchmark_main.cc
   standalone/benchmark_text_input.cc
   standalone/rcv1_benchmarks.cc
+  standalone/string_view_benchmarks.cc
```

Comment:
Though this is an interesting thing to see, this does not seem relevant to VW's performance. Adding this is going to make the benchmarks CI slower, which is already one of the slower runs.

Reply:
I ran the benchmarks to see how long they take.  It looks like this adds less than a second to the time to run the process.  However, I think the more important question you ask is what kind of benchmarks should we have?  Perhaps we can discuss this more during a VW session?

Reply:
We use the benchmarks CI to make a determination about the performance implications of a PR. From my perspective benchmarking `std::string_view` by ref or value is irrelevant to the goal of that CI and just introduces noise.

Reply:
Let's find a good place for it then.  We should be micro benchmarking small code blocks to help guide dev choices.

Reply:
The `std::string_view` benchmark was helpful in answering the question of value vs ref. Now that that is answered I do not see the value of persisting this benchmark. Ephemeral code to measure things is important, but persisting all of this code we write is not necessarily the right thing to do.

Benchmarks for components we develop and maintain are certainly helpful as they allow us to track performance over time. Benchmarks for components we do not develop are helpful to answer "this or that" questions but are not useful to track over time as they are beyond our control.

I'm all for micro-benchmarking of small code blocks we develop.

Reply:
There are a few very different points raised.  Let me summarize and address them here.

1) This microbenchmark answers the question of ref vs not.  Once there is an answer, we don't need to keep the benchmark around.
a) On the contrary, we should preserve these benchmarks that help make decisions so we don't have to revisit this question.  It makes perf knowledge institutional vs tribal.  

2) Benchmarks CI is used to determine perf implications of a PR.  This benchmark will not help with that goal.
a) VW::string_view is an alias for boost::string_view or boost::string_ref or could be something else in the future.  That might be an argument for leaving this where it is.

3) Benchmark CI takes a long time.  We should not be slowing it down further.
a) I measured the impact of this benchmark relative to the other ones.  It takes approx. 1 second. 

4) Persisting all of this code we write is not necessarily the right thing to do.
a) Yes.  I agree.

Reply:
If we want to document this decision the [coding guidelines](https://github.com/VowpalWabbit/vowpal_wabbit/wiki/Coding-guidelines) is the way to go here

I love that we benchmark these things but this is likely to be a useful benchmark when we (a) remove boost (b) update c++ version or (c) update cmake 

The time to run this might only be one second but developers will have the extra cognitive load of either learning to ignore this benchmark (never good to get into the mental state of ignoring things) in the CI run and/or opening the file to see what this benchmark does and why when all other benchmarks have gone up/down this one has stayed the same

If we want microbenchmarks of small code blocks then (in a separate PR from this one) we can create a different CI run or benchmark category that clearly states what these benchmarks are doing

Reply:
I am fine with another location in the repo for microbenchmarks used to justify coding decisions around performance.  I think this is essential if performance is important.  I can move this to a different PR.